File: LeetCode/9-PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        // Condizione Base
        if (x < 0) {
            return false;
        }
        // Salviamo il valore nella variabile "number"
        int number = x;
        // Variabile per leggere il numero reverse
        int reverse = 0;
        while (number > 0) {
            reverse = reverse * 10 + number % 10;
            number /= 10;
        }
        return x == reverse;
    }
}

File: LeetCode/9-PalindromeNumber.c
bool isPalindrome(int x) {
    // Condizione Base
    if (x < 0) return false;
    // Variabile Reverse
    long reverse = 0;
    // Variabile numero
    long numero=x;
    while (numero > 0) {
        reverse = reverse * 10 + numero % 10;
        numero /= 10;
    }
    return (reverse == x);
}